### PR TITLE
Load share signal

### PR DESCRIPTION
**Pull request by skarbat**
@tombettany this PR replaces this one: https://github.com/KanoComputing/sonic-pi/pull/44

I need to finish the unfolding of the shared filename into the workspace.
right now only the string is put into it.

cc @Ealdwulf

### CODE REVIEW

**Comment by Ealdwulf** [comment]
@tombettany Are you sure the version should not change? There has been an api change (new dbus method). @skarbat were you going to make some package depend on >=2.10.0?

**Comment by skarbat** [comment]
@Ealdwulf I wasn't making any package depend on the new version, but simply adding it for clarifying reasons as you suggest.

**Comment by Ealdwulf** [comment]
Okay, then I don't mind whether it's bumped.

**Comment by tombettany** [comment]
Also @Ealdwulf we bumped the version last week so we haven't released it anyway.

**Comment by skarbat** [comment]
@tombettany @Ealdwulf I have added the functionality to load the shared code file into the workspace.
Works well on the RPi. Does it look ok to you?

File: app/gui/qt/mainwindow.cpp [hunk 1]
```diff
@@ -1262,6 +1262,38 @@ void MainWindow::load()
...
+    if (!sharedCreation.exists() || !sharedCreation.open(QIODevice::ReadOnly | QIODevice::Text)) {
+      QMessageBox load_failed;
+      load_failed.setText(QString(tr("The Shared creation file could not be loaded: %1").arg(share_filename)));
+      load_failed.exec();
+    }
+    else {
+      // Ask confirmation before removing the current creation
+      QMessageBox::StandardButton reply;
+      reply = QMessageBox::question(this, tr("Loading a share"),
+				    tr("Are you sure you want to load this share? You will loose the current workspace"),
```
Comment (tombettany) [inline app/gui/qt/mainwindow.cpp]: Should be `lose` not `loose`.

File: app/gui/qt/mainwindow.cpp [hunk 2]
```diff
@@ -1262,6 +1262,38 @@ void MainWindow::load()
...
+				    QMessageBox::Yes|QMessageBox::No);
+
+      if (reply == QMessageBox::Yes) {
+	QTextStream shared_code(&sharedCreation);
+	getCurrentWorkspace()->setText(sharedCreation.readAll());
+	sharedCreation.close();
```
Comment (tombettany) [inline app/gui/qt/mainwindow.cpp]: Should we not close the file regardless of if they click yes?

**Comment by tombettany** [comment]
Other than some minor points, it looks good.

**Comment by skarbat** [comment]
thank you Tom! I fixed your remarks. @tombettany

**Pull request merged**